app/controllers/organizations_controller.rb

```diff
@@ -1,5 +1,46 @@
 class OrganizationsController < ApplicationController
-  def show
+  before_action :authenticate_org_admin, only: %i[edit update]
+  before_action :find_organization, only: %i[show edit update]
+
+  skip_before_action :authenticate_user!, only: [:show_on_subdomain]
+
+  def index
+    @organizations = current_user.organizations
+    @user_organizations = UserOrganization.all.where(user_id: current_user.id)
+  end
+
+  def update
+    if @organization.update_attributes(organization_params)
+      redirect_to action: 'index'
+    else
+      render :edit
+    end
+  end
+
+  def show_on_subdomain
     @organization = Organization.find_by subdomain: request.subdomain
+    render :show
+  end
+
+
+  private
+
+  def authenticate_org_admin
```

Comment:
Use full words: `organization` instead of `org`
Simplify this method here and below (the simplest, the better):

```ruby
def authenticate_organization_admin
  if !user.user_organizations.exists?(role: 'admin', organization_id: params[:id])
    render file: 'public/403.html'
  end
end
```

Challenge question: will this cause a double render error?